src/main/java/org/jitsi/videobridge/LipSyncHack.java

```diff
@@ -434,6 +445,107 @@ public void onRTPTranslatorWillWriteVideo(
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTPTransformer()
+    {
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer()
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close()
+    {
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RawPacket[] reverseTransform(RawPacket[] pkts)
+    {
+        return pkts;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RawPacket[] transform(RawPacket[] pkts)
+    {
+        if (pkts == null || pkts.length == 0)
+        {
+            return pkts;
+        }
+
+        RawPacket[] extra = null;
+
+        for (int i = 0; i < pkts.length; i++)
+        {
+            if (pkts[i] == null)
+            {
+                continue;
+            }
+
+            long ssrc = pkts[i].getSSRCAsLong();
+            if (!ssrcsWithoutBlackKeyframes.isEmpty()
```

Comment:
That's right. If the participant joins video muted/hidden, then we inject black key frames with the inject task. If the participant joins with his video on, then the inject task never kicks in because it sees video packets, and that's when the transformer comes into play.